### ISSUE TITLE
Perf rec seems to be messing up later annotates

### ISSUE THREAD

**Issue by tolomea** [adamchainz/django-perf-rec#322]
This fails with `Cannot resolve keyword 'submitted' into field.`

```
from django.test import TestCase
from django_perf_rec import TestCaseMixin

class TestExportLogic(TestCaseMixin, TestCase):
    def test_b(self):
        with self.record_performance():
            pass

        models.Row.objects.annotate(
            submitted=F("state"),
            for_ron=F("submitted"),
        )
```

state is a legit field on Row

If you comment out the record_performance bit it passes.

Django==3.1.4
django-perf-rec==4.10.0
Python 3.8.6
psycopg2==2.8.6

**Comment by tolomea** [comment]
```
python manage.py test approval.tests.TestExportLogic
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
E
======================================================================
ERROR: test_b (approval.tests.TestExportLogic)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "<redacted>\tests.py", line 2127, in test_b
    models.Row.objects.annotate(
  File "<redacted>\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "<patchy>", line 36, in annotate
  File "<redacted>\site-packages\django\db\models\sql\query.py", line 1018, in add_annotation
    annotation = annotation.resolve_expression(self, allow_joins=True, reuse=None,
  File "<redacted>\site-packages\django\db\models\expressions.py", line 539, in resolve_expression
    return query.resolve_ref(self.name, allow_joins, reuse, summarize)
  File "<redacted>\site-packages\django\db\models\sql\query.py", line 1683, in resolve_ref
    join_info = self.setup_joins(field_list, self.get_meta(), self.get_initial_alias(), can_reuse=reuse)
  File "<redacted>\site-packages\django\db\models\sql\query.py", line 1565, in setup_joins
    path, final_field, targets, rest = self.names_to_path(
  File "<redacted>\site-packages\django\db\models\sql\query.py", line 1481, in names_to_path
    raise FieldError("Cannot resolve keyword '%s' into field. "
django.core.exceptions.FieldError: Cannot resolve keyword 'submitted' into field. Choices are: applicants, created_at, data_export, data_export_id, data_import, data_import_id, firm, firm_id, id, issues, mortgage, original_applicants, original_mortgage, original_property, property, row_id, state, updated_at

```

**Comment by adamchainz** [comment]
Fixed in 4.11.0.
